### PR TITLE
adding option to pass in a json blob with parameter options into playdoh

### PR DESCRIPTION
**Pull request by shirchen**
this is my first nodejs code ever! so please tell me if I'm out of line on anything.

### CODE REVIEW

**Comment by Raynos** [comment]
What's the use case and what's the new public interface ? 

Can you update the README ?

File: index.js [hunk 1]
```diff
@@ -39,6 +39,13 @@ function Template(name, opts) {
         if (e.code !== 'MODULE_NOT_FOUND') { throw e; }
         this.exclusions = [];
     }
+    this.fromJson = false;
+    // Allowing overriding of values to be passed in instead of gathering them from user input.
+    if (opts.jsonValues.length && opts.jsonValues.length > 0) {
```
Comment (Raynos) [inline index.js]: this is not how you check for the size of an object.

Use `Object.keys(opts.jsonValues).length` instead.

File: index.js [hunk 2]
```diff
@@ -53,9 +60,6 @@ Template.prototype.init = function(dest, callback) {
     var vars = self.mod;
     var keys = Object.keys(vars);
 
-    if (dest) {
```
Comment (Raynos) [inline index.js]: why did this line move ?

**Comment by Raynos** [comment]
This looks reasonable.

Can we update README & bin/usage.md in this PR ?

File: index.js (hunk 1 above)
Comment (shirchen) [inline index.js]: done.

File: index.js (hunk 2 above)
Comment (shirchen) [inline index.js]: i was initially checking if self.values was empty but then switched to just having a flag fromJson. putting back.

**Comment by sh1mmer** [comment]
This looks :candy: 

:+1: from me

**Pull request merged**